error-codes/README.md

```diff
@@ -0,0 +1,97 @@
+# Error Codes
+
+## Introduction
+In the event that a node detects violation of a protocol or is unable to
+complete the necessary steps required for the protocol, it's useful to provide a
+reason for disconnection to the other end. This error code can be sent on both
+Connection Close and Stream Reset. Its purpose is similar to http response
+status. A server, on receiving an invalid request can reset the stream providing
+a `BAD_REQUEST` error code, when it's busy handling too many requests can
+provide a `RATE_LIMITED` error code, etc. An error code doesn't always indicate
+an error condition. For example, a connection may be closed prematurely because
+a connection to the same peer on a better transport is available. 
+
+## Semantics
+Error codes are unsigned 32bit integers. The range 0 to 10000 is reserved for
+libp2p errors. Application specific errors can be defined by protocols from
+integers outside of this range. Implementations supporting error codes MUST
+provide the error code provided by the other end to the application.
+
+Error codes provide a best effort guarantee that the error will be propogated to
+the applciation layer. This provides backwards compatibility with older nodes,
+allows smaller implementations and using transports that don't provide a
```

Comment:
I'm not sure about popularity here - maybe "simpler" instead of "smaller"?

Either way this needs an Oxford comma:

```suggestion
allows smaller implementations, and using transports that don't provide a
```